lib/backend/k8s/custom/globalbgppeer.go

```diff
@@ -12,76 +12,85 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package thirdparty
+package custom
 
 import (
 	"encoding/json"
 
 	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/net"
+	"github.com/projectcalico/libcalico-go/lib/scope"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
-// GlobalBgpPeer is the ThirdPartyResource definition of a Calico Global BGP Peer resource in
+// BGPPeer is the CustomResourceDefinition of a Calico BGP Peer resource in
 // the Kubernetes API.
-type GlobalBgpPeer struct {
+type BGPPeer struct {
 	metav1.TypeMeta `json:",inline"`
 	Metadata        metav1.ObjectMeta `json:"metadata"`
-	Spec            api.BGPPeerSpec   `json:"spec"`
+	Spec            BGPPeerSpec       `json:"spec"`
 }
 
-// GlobalBgpPeerList is a list of Calico Global BGP Peer resources.
-type GlobalBgpPeerList struct {
+type BGPPeerSpec struct {
+	Spec   api.BGPPeerSpec
```

Comment:
but then in the conversion code, we will have to assign individual fields from api Spec when we do ToKVPair, which will need an update every time we change the spec. https://github.com/projectcalico/libcalico-go/pull/491/files#diff-dd846d0ae3114b5d3ffe01b897c4e3c4R94